contracts/test/integration/vip/vip7.sol

```diff
@@ -0,0 +1,161 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity =0.8.13;
+
+import {TimelockController} from "@openzeppelin/contracts/governance/TimelockController.sol";
+import {TimelockSimulation} from "../utils/TimelockSimulation.sol";
+import {MainnetAddresses} from "../fixtures/MainnetAddresses.sol";
+import {DSTest} from "./../../unit/utils/DSTest.sol";
+import {Core} from "../../../core/Core.sol";
+import {Vm} from "./../../unit/utils/Vm.sol";
+import {IVIP} from "./IVIP.sol";
+import {AllRoles} from "./../utils/AllRoles.sol";
+import {Volt} from "../../../volt/Volt.sol";
+import {PriceBoundPSM} from "../../../peg/PriceBoundPSM.sol";
+import {PCVGuardian} from "../../../pcv/PCVGuardian.sol";
+import {MakerRouter} from "../../../pcv/maker/MakerRouter.sol";
+
+contract vip7 is DSTest, IVIP, AllRoles {
+    Vm public constant vm = Vm(HEVM_ADDRESS);
+
+    function getMainnetProposal()
+        public
+        pure
+        override
+        returns (TimelockSimulation.action[] memory proposal)
+    {
+        proposal = new TimelockSimulation.action[](4);
+
+        proposal[0].target = MainnetAddresses.VOLT_FEI_PSM;
+        proposal[0].value = 0;
+        proposal[0].arguments = abi.encodeWithSignature("pauseMint()");
+        proposal[0].description = "Pause Minting on the FEI PSM";
+
+        proposal[1].target = MainnetAddresses.PCV_GUARDIAN;
+        proposal[1].value = 0;
+        proposal[1].arguments = abi.encodeWithSignature(
+            "withdrawAllERC20ToSafeAddress(address,address)",
+            MainnetAddresses.VOLT_FEI_PSM,
+            MainnetAddresses.VOLT
+        );
+        proposal[1].description = "Remove all VOLT from FEI PSM";
+
+        proposal[2].target = MainnetAddresses.PCV_GUARDIAN;
+        proposal[2].value = 0;
+        proposal[2].arguments = abi.encodeWithSignature(
+            "addWhitelistAddress(address)",
+            MainnetAddresses.VOLT_DAI_PSM
+        );
+        proposal[2]
+            .description = "Add DAI PSM to whitelisted addresses on PCV Guardian";
+
+        proposal[3].target = MainnetAddresses.VOLT_USDC_PSM;
+        proposal[3].value = 0;
+        proposal[3].arguments = abi.encodeWithSignature("unpauseRedeem()");
+        proposal[3].description = "Unpause redemptions for USDC PSM";
+    }
+
+    function mainnetSetup() public override {}
```

Comment:
setup should transfer some VOLT to the new DAI PSM because that is an action the multisig will likely need to take.